### PR TITLE
[Fix doc issue 6058] Add consumer descriptions

### PR DESCRIPTION
**Pull request by Anonymitaet**
Fix https://github.com/apache/pulsar/issues/6058

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@sijie could u pls help review? thanks

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -55,6 +55,8 @@ If batching is enabled, the producer will accumulate and send a batch of message
 
 A consumer is a process that attaches to a topic via a subscription and then receives messages.
 
+A consumer pulls messages from a broker. There is a queue at the consumer side to receive messages pushed from the broker. The queue size is configurable by [`receiverQueueSize`](client-libraries-java.md#configure-consumer) (default: 1000). Each time `consumer.receive()` is called, a message is dequeued from the buffer. If the requested message size of a consumer is larger than the queue size, after all messages in the queue are consumed, messages are delivered to the consumer immediately once there comes new messages in the queue.
```
Comment (codelipenghui) [inline site2/docs/concepts-messaging.md]: > If the requested message size of a consumer is larger than the queue size, after all messages in the queue are consumed, messages are delivered to the consumer immediately once there comes new messages in the queue.  

Consumer will pull more messages while available messages in the receive queue < half of the queue size.

**Comment by Anonymitaet** [comment]
@codelipenghui Thanks for your guidance, I've incorporated your comments as we discussed, PTAL

**Comment by codelipenghui** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Pull request merged**